packages/vscode/src/plugins/nextjs/findNodes.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ArrowFunction,
+  FunctionDeclaration,
+  SourceFile,
+  TypeGuards,
+} from 'ts-morph'
+import { NextFunctionType } from './constants'
+/**
+ * This finds the relevant ast nodes to add the types to
+ * @param sourceFile
+ * @param foundNextFunctions
+ * @param callback
+ */
+export function findNodes(
+  sourceFile: SourceFile,
+  foundNextFunctions: NextFunctionType,
+  addTypes: (
+    node: ArrowFunction | FunctionDeclaration,
+    foundNextFunctions: NextFunctionType,
+  ) => void,
+): void {
+  const defaultExportSymbol = sourceFile.getDefaultExportSymbol()
+  if (defaultExportSymbol) {
+    // you will need to handle more scenarios than what's shown here
+    const declaration = defaultExportSymbol.getDeclarations()[0]
```

Comment:
there should be a check here before accessing an element of the array

Reply:
If the array is empty i.e `const this = []`  calling `this[0]` will not throw an error but will return undefined